app/routes/wow.quantity-manipulation.tsx

```
@@ -0,0 +1,396 @@
+  const response = await WoWQuantityManipulation(validInput.data)
+  const data = await response.json()
+
+  if (!response.ok) {
+    const errorData = data as ErrorResponse
+    return json({ exception: JSON.stringify(errorData, null, 2) })
+  }
+
+  return json(data)
+}
+
+type ActionResponse = ManipulationResponse | { exception: string } | {}
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)